client/src/app/services/mentorship-request.service.ts

```diff
@@ -0,0 +1,20 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { MentorshipRequestCreateDTO } from '../interfaces/mentorship-request';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MentorshipRequestService {
+  constructor(private httpService: HttpClient) {}
+
+  async requestMentorship(
+    body: MentorshipRequestCreateDTO
+  ): Promise<Observable<MentorshipRequestCreateDTO>> {
+    return this.httpService.post<MentorshipRequestCreateDTO>(
```

Comment:
nit: why is this `Promise<Observable<...>>`? Does that type-check for you? It doesn't seem to if I try it on my end.

Reply:
I will change later